File: app/src/main/java/com/example/splitpayandroid/annotation/Annotations.java
package com.example.splitpayandroid.annotation;

import dagger.MapKey;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

enum MessageType {
    TEXT_MESSAGE_TYPE,
    IMAGE_MESSAGE_TYPE,
    INFO_MESSAGE_TYPE;
}



@Target({ElementType.METHOD})
@Retention(RetentionPolicy.RUNTIME)
@MapKey
@interface Annotations {
    MessageType value();
}

File: app/src/main/java/com/example/splitpayandroid/annotation/annotations.kt
package com.example.splitpayandroid.annotation

import dagger.MapKey


@Target(AnnotationTarget.CLASS, AnnotationTarget.FUNCTION,
    AnnotationTarget.VALUE_PARAMETER, AnnotationTarget.EXPRESSION, AnnotationTarget.PROPERTY, AnnotationTarget.FIELD
)
@Retention(AnnotationRetention.SOURCE)
@MustBeDocumented
@javax.inject.Qualifier
annotation class FirstMap


@Target(AnnotationTarget.CLASS, AnnotationTarget.FUNCTION,
    AnnotationTarget.VALUE_PARAMETER, AnnotationTarget.EXPRESSION, AnnotationTarget.PROPERTY, AnnotationTarget.FIELD
)
@Retention(AnnotationRetention.SOURCE)
@MustBeDocumented
@javax.inject.Qualifier
annotation class SecondMap

enum class MessageType {
    TEXT_MESSAGE_TYPE,
    IMAGE_MESSAGE_TYPE,
    INFO_MESSAGE_TYPE
}

@Target(AnnotationTarget.FUNCTION, AnnotationTarget.FIELD)
@Retention(AnnotationRetention.RUNTIME)
@MapKey
annotation class ProviderKey(val messageType: MessageType)